src/image/prepare_single_image_build_matrix.py

```diff
@@ -4,11 +4,61 @@
 from datetime import datetime, timezone
 import glob
 import json
-import os
 import pydantic
 import yaml
-
-from utils.schema.triggers import ImageSchema
+from typing import Any
+from pathlib import Path
+from git import Repo
+from tempfile import TemporaryDirectory as tempdir
+from copy import deepcopy
+import logging
+
+
+from src.uploads.infer_image_track import get_base_and_track
+from src.shared.github_output import GithubOutput
+from src.image.utils.schema.triggers import ImageSchema
+
+# TODO:
```

Comment:
Good point. I think it does. We might need some sort of validator to check the `source` and transform it to a GH URL when appropriate only. 

It's what you already do in the custom checkout action. So the question is whether we can transform this data in a single place, early in the pipeline, instead of duplicating the logic.

Reply:
I created a ticket for tracking.  https://warthogs.atlassian.net/browse/ROCKS-1613 

We can check/transform the `source` strings just after the trigger yaml is loaded, but I think we should keep the existing check/transform in the Build-Rock workflow for ease of use by others. The existing transform logic in the checkout action could be made callable by python so we prevent duplicating the logic.